test/foundry/Overflows.t.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: CAL
+pragma solidity =0.8.17;
+
+import {CreateOffchainAssetReceiptVaultFactory} from "../../contracts/test/CreateOffchainAssetReceiptVaultFactory.sol";
+import {Test, Vm, stdError} from "forge-std/Test.sol";
+import {OffchainAssetReceiptVault} from "../../contracts/vault/offchainAsset/OffchainAssetReceiptVault.sol";
+import {OffchainAssetVaultCreator} from "./OffchainAssetVaultCreator.sol";
+
+contract Overflows is Test, CreateOffchainAssetReceiptVaultFactory {
+    /// Check positive overflow for mint
+    function testMintOverflow(
+        uint256 fuzzedKeyAlice,
+        bytes memory fuzzedReceiptInformation,
+        string memory assetName,
+        string memory assetSymbol
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+
+        // Prank as Alice for the transaction
+        vm.startPrank(alice);
+
+        vault.grantRole(vault.DEPOSITOR(), alice);
+
+        vm.expectRevert(stdError.arithmeticError);
+        vault.mint(UINT256_MAX + 1, alice, 1e18, fuzzedReceiptInformation);
```

Comment:
3 things:

- `type(uint256).max` is already a constant in solidity you should use that
- the error of adding `1` to `type(uint256).max` is happening _before mint is called_ so you're not testing anything here
- maybe we can fuzz this 1e18 value here